web/views/teacher.py

```diff
@@ -322,6 +323,22 @@ def student_list(request):
     return render(request, "web/student_list.html")
 
 
+@user_passes_test(is_teacher)
+def student_page(request, login: int):
+    """
+    Page that renders a Vue component with a detail of a single student.
+    """
+    student = get_object_or_404(User, username=login)
```

Comment:
In the case I enter it manually in lowercase.

```suggestion
    login = login.upper()
    student = get_object_or_404(User, username=login)
```